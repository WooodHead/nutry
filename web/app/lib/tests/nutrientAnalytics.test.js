import { Map } from 'immutable';
import moxios from 'moxios';
import sinon from 'sinon';
import {
  getFoodProfileMockSuccessFull } from 'mocks/getFoodProfileMock';
import {
  getSearchParsedSuccess,
  getSearchParsedSuccessMulti } from 'mocks/getElasticsearchQueryMock';
import {
  getFilteredData,
  getRankingResults,
} from '../nutrientAnalytics';


describe('nutrientAnalytics ', () => {
  describe('getFilteredData', () => {
    it('selects filtered data', () => {
      const nutrients = Map({
        CHOCDF: Map({
          name: 'Fatty acids, total polyunsaturated',
          units: 'g',
          value: 1.19,
          selected: false }),
        VITA: Map({
          name: 'Retinol Equivalents (VIT A)',
          units: 'ug',
          value: 190,
          selected: false }),
        FOLDFE: Map({
          name: 'Dietary folate equivalents',
          units: 'ug',
          value: 274,
          selected: false }) });
      const nutrientFilter = ['FAPU'];
      const ageGroupSelected = Map({ value: 'AM19', label: 'Adult Male (19-30)', className: 'am-19' });
      expect(getFilteredData(nutrients, nutrientFilter, ageGroupSelected)).toMatchSnapshot();
    });
    it('handles empty data', () => {
      const ageGroupSelected = Map({ value: 'AM19', label: 'Adult Male (19-30)', className: 'am-19' });
      expect(getFilteredData(undefined, undefined, ageGroupSelected)).toEqual([]);
    });
  });
  describe('getRankingResults', () => {
    beforeEach(() => {
      moxios.install();
    });
    afterEach(() => {
      moxios.uninstall();
    });
    it('processes searchResults', (done) => {
      const onFulfilled = sinon.spy();
      moxios.stubRequest(/.*profiles\/02049.json.*/, getFoodProfileMockSuccessFull);
      getRankingResults(getSearchParsedSuccess).then(onFulfilled);
      moxios.wait(() => {
        expect(onFulfilled.getCall(0).args[0].toJS()).toMatchSnapshot();
        done();
      });
    });
    it('processes searchResults when multiple results', (done) => {
      const onFulfilled = sinon.spy();
      moxios.stubRequest(/.*profiles\/02049.json.*/, getFoodProfileMockSuccessFull);
      moxios.stubRequest(/.*profiles\/11111.json.*/, getFoodProfileMockSuccessFull);
      getRankingResults(getSearchParsedSuccessMulti).then(onFulfilled);
      moxios.wait(() => {
        expect(onFulfilled.getCall(0).args[0].toJS()).toMatchSnapshot();
        done();
      });
    });
    it('processes empty searchResults', (done) => {
      // moxios.stubRequest(/.*profiles\/02049.json.*/, {});
      expect(getRankingResults({})).toEqual([]);
      done();
    });
  });
});
