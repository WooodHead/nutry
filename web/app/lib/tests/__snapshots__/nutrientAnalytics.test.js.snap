// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nutrientAnalytics  getFilteredData selects filtered data 1`] = `
Immutable.List [
  Immutable.Map {
    "prefix": "FAPU",
    "name": "Fatty acids, total polyunsaturated",
    "units": "g",
    "value": "~",
    "rdi": null,
  },
]
`;

exports[`nutrientAnalytics  getRankingResults processes searchResults 1`] = `
Object {
  "ALA": Array [],
  "ARG": Array [],
  "ASP": Array [],
  "BETN": Array [],
  "CA": Array [],
  "CARTA": Array [],
  "CARTB": Array [],
  "CHOCDF": Array [
    Immutable.Map {
      "name": "Fresh thyme",
      "id": "02049",
      "value": 1,
      "unit": "g",
    },
  ],
  "CHOLN": Array [],
  "CRYPX": Array [],
  "CU": Array [],
  "CYS": Array [],
  "ENERC": Array [],
  "ENERC_KCAL": Array [],
  "FAMS": Array [],
  "FAPU": Array [],
  "FASAT": Array [],
  "FAT": Array [],
  "FATRN": Array [],
  "FE": Array [],
  "FIBTG": Array [],
  "FLD": Array [],
  "FOL": Array [],
  "FOLAC": Array [],
  "FOLDFE": Array [],
  "FOLFD": Array [],
  "FRUS": Array [],
  "GALS": Array [],
  "GLU": Array [],
  "GLUS": Array [],
  "HISTN": Array [],
  "HYP": Array [],
  "ILE": Array [],
  "K": Array [],
  "LACS": Array [],
  "LCW3TOTAL": Array [],
  "LEU": Array [],
  "LUT+ZEA": Array [],
  "LYCPN": Array [],
  "LYS": Array [],
  "MALS": Array [],
  "MET": Array [],
  "MG": Array [],
  "MN": Array [],
  "NA": Array [],
  "NIA": Array [],
  "P": Array [],
  "PHE": Array [],
  "PRO": Array [],
  "PROCNT": Array [],
  "RETOL": Array [],
  "SE": Array [],
  "SER": Array [],
  "SUGAR": Array [],
  "THIA": Array [],
  "THR": Array [],
  "TOCPHA": Array [],
  "TOCPHB": Array [],
  "TOCPHD": Array [],
  "TRP": Array [],
  "TYR": Array [],
  "VAL": Array [],
  "VITA_IU": Array [],
  "VITA_RAE": Array [],
  "VITB6A": Array [],
  "VITC": Array [],
  "VITD": Array [],
  "VITK": Array [],
  "WATER": Array [
    Immutable.Map {
      "name": "Fresh thyme",
      "id": "02049",
      "value": 0,
      "unit": "g",
    },
  ],
  "ZN": Array [],
}
`;

exports[`nutrientAnalytics  getRankingResults processes searchResults when multiple results 1`] = `
Object {
  "ALA": Array [],
  "ARG": Array [],
  "ASP": Array [],
  "BETN": Array [],
  "CA": Array [],
  "CARTA": Array [],
  "CARTB": Array [],
  "CHOCDF": Array [
    Immutable.Map {
      "name": "Fresh thyme",
      "id": "11111",
      "value": 1,
      "unit": "g",
    },
    Immutable.Map {
      "name": "Fresh thyme",
      "id": "02049",
      "value": 1,
      "unit": "g",
    },
  ],
  "CHOLN": Array [],
  "CRYPX": Array [],
  "CU": Array [],
  "CYS": Array [],
  "ENERC": Array [],
  "ENERC_KCAL": Array [],
  "FAMS": Array [],
  "FAPU": Array [],
  "FASAT": Array [],
  "FAT": Array [],
  "FATRN": Array [],
  "FE": Array [],
  "FIBTG": Array [],
  "FLD": Array [],
  "FOL": Array [],
  "FOLAC": Array [],
  "FOLDFE": Array [],
  "FOLFD": Array [],
  "FRUS": Array [],
  "GALS": Array [],
  "GLU": Array [],
  "GLUS": Array [],
  "HISTN": Array [],
  "HYP": Array [],
  "ILE": Array [],
  "K": Array [],
  "LACS": Array [],
  "LCW3TOTAL": Array [],
  "LEU": Array [],
  "LUT+ZEA": Array [],
  "LYCPN": Array [],
  "LYS": Array [],
  "MALS": Array [],
  "MET": Array [],
  "MG": Array [],
  "MN": Array [],
  "NA": Array [],
  "NIA": Array [],
  "P": Array [],
  "PHE": Array [],
  "PRO": Array [],
  "PROCNT": Array [],
  "RETOL": Array [],
  "SE": Array [],
  "SER": Array [],
  "SUGAR": Array [],
  "THIA": Array [],
  "THR": Array [],
  "TOCPHA": Array [],
  "TOCPHB": Array [],
  "TOCPHD": Array [],
  "TRP": Array [],
  "TYR": Array [],
  "VAL": Array [],
  "VITA_IU": Array [],
  "VITA_RAE": Array [],
  "VITB6A": Array [],
  "VITC": Array [],
  "VITD": Array [],
  "VITK": Array [],
  "WATER": Array [
    Immutable.Map {
      "name": "Fresh thyme",
      "id": "11111",
      "value": 0,
      "unit": "g",
    },
    Immutable.Map {
      "name": "Fresh thyme",
      "id": "02049",
      "value": 0,
      "unit": "g",
    },
  ],
  "ZN": Array [],
}
`;
